source/common/tcp_proxy/upstream.h

```diff
@@ -221,21 +528,43 @@ class HttpUpstream : public GenericUpstream, protected Http::StreamCallbacks {
 
 class Http1Upstream : public HttpUpstream {
 public:
-  Http1Upstream(Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
+  Http1Upstream(HttpConnPool& http_conn_pool, Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
+                Http::StreamDecoderFilterCallbacks& decoder_callbacks, Router::Route& route,
                 const TunnelingConfigHelper& config, StreamInfo::StreamInfo& downstream_info);
-
+  ~Http1Upstream() override;
   void encodeData(Buffer::Instance& data, bool end_stream) override;
   void setRequestEncoder(Http::RequestEncoder& request_encoder, bool is_ssl) override;
   bool isValidResponse(const Http::ResponseHeaderMap& headers) override;
+  void newStream(GenericConnectionPoolCallbacks&) override {}
+  void setRouterUpstreamRequest(UpstreamRequestPtr) override {}
 };
 
 class Http2Upstream : public HttpUpstream {
 public:
-  Http2Upstream(Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
+  Http2Upstream(HttpConnPool& http_conn_pool, Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
+                Http::StreamDecoderFilterCallbacks& decoder_callbacks, Router::Route& route,
                 const TunnelingConfigHelper& config, StreamInfo::StreamInfo& downstream_info);
-
+  ~Http2Upstream() override;
   void setRequestEncoder(Http::RequestEncoder& request_encoder, bool is_ssl) override;
   bool isValidResponse(const Http::ResponseHeaderMap& headers) override;
+  void newStream(GenericConnectionPoolCallbacks&) override {}
+  void setRouterUpstreamRequest(UpstreamRequestPtr) override {}
+};
+
+class CombinedUpstream : public HttpUpstream {
```

Comment:
Please add a descriptive comment which explains what this class is all about.

Reply:
done